clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -190,6 +197,21 @@ opcode_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         memref.data.type != TRACE_TYPE_INSTR_NO_FETCH) {
         return true;
     }
+
+    /* At this point we start processing memref instructions, so we're past the header of
+     * the trace, which contains the trace filetype. If we didn't encounter a
+     * TRACE_MARKER_TYPE_FILETYPE we return an error and stop processing the trace.
+     * XXX i#6812: we could allow traces that have some shards with no filetype, as long
+     * as there is at least one shard with it, by caching the filetype from shards that
+     * have it and using that one. We can do this using memtrace_stream_t::get_filetype(),
+     * which requires updating opcode_mix to use parallel_shard_init_stream() rather than
+     * the current (and deprecated) parallel_shard_init().
+     */
+    if (shard->filetype == OFFLINE_FILE_TYPE_DEFAULT) {
```

Comment:
Maybe add a comment by the declaration of OFFLINE_FILE_TYPE_DEFAULT that says that we always expect the filetype to contain one of the other enums too (at least the arch e.g.), and so OFFLINE_FILE_TYPE_DEFAULT should never be present alone, and can be used an an error check.

Reply:
Done.